File: 0112-path-sum/0112-path-sum.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def hasPathSum(self, root: Optional[TreeNode], targetSum: int) -> bool:
        if not root:
            return False
        
        a = targetSum - root.val
        
        if not root.left and not root.right:
            if a == 0:
                return True
            else:
                 return False
        b = False
        c = False
        if root.left:
          b =  self.hasPathSum(root.left, a)
        if root.right:
           c = self.hasPathSum(root.right, a)
        return b or c
        

File: 0112-path-sum/0112-path-sum.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public boolean hasPathSum(TreeNode root, int targetSum) {
        if(root == null){
            return false;
        }
        int a = targetSum - root.val;

        if(root.left == null && root.right == null){
            if(a == 0){
                return true;
            }
            else{
                return false;
            }
        }
        boolean c = false;
        boolean d = false;

        if(root.left != null){
            c = hasPathSum(root.left, a);
        }
        
        if(root.right != null){
            d = hasPathSum(root.right, a);
        }

        return c || d;
    }
}